### DISCUSSION TITLE
Alpine Expression Error: e.content is undefined

### DISCUSSION THREAD

**Discussion by cognivore** [alpinejs/alpine#2666] [4. General]
Hi, I'm trying to build a thing which will conditionally show either login/register forms or greet the user by their name. Below is the code with a relevant import.

When I try to run it, I get a rather cryptic errors, attached after the snippet. The gist is that every time I try to use `x-if` with logical not (`!`), it throws an error, whereas using it without `!` just doesn't work, silently. I can, however, access to the whole state and inspect it, say with `console.log()` from withing `x-if`s.

I'm not quite sure what's the story here, since it's an absolutely basic pattern I was using elswhere, and I feel it has never failed. I would appreciate if someone would explain what's the actual fundamental issue with the code. Thanks!

```
    <section x-data="{
        password: '',
        name: 'come on alpine, not now',
        slip: '',
        signingKeypair: {
            public: '',
            secret: ''
        }
    }">
        <h1>DoAuth-based authentication</h1>
        <section>
            <div>please, alpine</div>
            <div x-text="name"></div>
            <div>come on, really</div>
            <div x-if="!signingKeypair['puiblic']">
                <section>
                    <h2>Register</h2>
                    <form x-on:submit="
                        $event.preventDefault();
                        signingKeypair = await mkIdentity(password);
                        console.log('Registration result: ', await register( signingKeypair,
                        grecaptcha.getResponse(grecaptcha_widgets['register--captcha']), {name: name} ));
                    ">
                        <label>Name: </label><input x-model="name">
                        <label>Password: </label><input x-model="password" type="password">
                        <div id="register--captcha"></div>
                        <input type="submit" value="register">
                    </form>
                </section>
                <section>
                    <h2>Login</h2>
                    <form x-on:submit="
                        $event.preventDefault();
                        signingKeypair = await mkIdentity(password);
                    ">
                        <label>Password: </label><input x-model="password" type="password">
                        <input type="submit" value="login">
                    </form>
                </section>
            </div>
            <div x-if="signingKeypair['puiblic']">
                <div>Hello, <span x-text="name"></span>!</div>
            </div>
        </section>
    </section>

    <script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>
    <script src="https://www.google.com/recaptcha/api.js?onload=recaptchaHandler&render=explicit" async defer></script>
```

**Comment by SimoTod** [comment]
Is that the real code? If yes, you misspelled `public` in your x-if condition

**Reply by cognivore** [reply]
Interesting. Gonna go fix that. I tried different approaches many many times, including dererring to "name", but yeah I'm gonna go fix the spelling mistake and see if it fixes things.

**Reply by cognivore** [reply]
Same errors: 
![image](https://user-images.githubusercontent.com/66186054/153729880-710d85c8-d6f1-4b1b-bdb0-86c3a5ef5ac5.png)

But thanks for finding a typo. I guess it's the "anger" part of "debugging in anger" :D

**Comment by SimoTod** [comment]
I didn't notice it initially on your example but X-if only works with template elements which I think it's the reason for the error. See https://alpinejs.dev/directives/if

**Reply by cognivore** [reply]
Thank you so much! It's been a while since I last used Alpine and I confused x-if with x-show. The annoying bit was, of course, that x-if would still make function calls and the lack of informative error message :)

Anyway, huge thanks, I fixed it instantaneously after reading your answer.
